File: util/perf/templates/test-template.js
/*
 *  {{TEST_DESCRIPTION}}
 */

if (typeof window == 'undefined') window = {}
if (!window.performance) window.performance = {}
window.performance.now = window.performance.now || Date.now

if (typeof print != 'undefined') log = print
else log = console.log

function test () {
  var start
  var duration

  // setup test

  start = window.performance.now()

  // begin test
  // ...
  // end test

  duration = window.performance.now() - start

  log(duration)
}

test()

File: util/perf/templates/test-template.c
/*
 * {{TEST_DESCRIPTION}}
 */

#include "stdio.h"
#include "time.h"

int main(int argc, char **argv) {
     int msec;
     clock_t start, duration;

     // setup test

     start = clock();

     // begin test
     // ...
     // end test

     duration = clock() - start;
     msec = duration * 1000 / CLOCKS_PER_SEC;

     printf("%d\n", msec);

     return 0;
}
